### DISCUSSION TITLE
Ensemble parallelism

### DISCUSSION THREAD

**Discussion by joernwichmann** [firedrakeproject/firedrake#3381] [Firedrake support] [answered]
I am playing with Ensemble parallelism, and I run this basic example to observe how local/global ranks work.

![Screenshot from 2024-02-02 12-52-06](https://github.com/firedrakeproject/firedrake/assets/116003022/efe69010-46cf-4966-914f-ff4673265d99)

Running this script with `mpiexec -n 4 ...` yields the following 

![Screenshot from 2024-02-02 12-53-26](https://github.com/firedrakeproject/firedrake/assets/116003022/939f8f11-d5d3-42ab-a292-718312921e70)

The console log warns about possible deadlocks. How do I prevent deadlocks from happening?

**Comment by connorjward** [comment]
First please share code using code blocks (inside backticks) rather than screenshots. It makes the code harder to reproduce if we were to want to run it ourselves.

How up-to-date is your Firedrake? @JDBetteridge recently made some changes to this teardown code.

**Comment by colinjcotter** [comment]
@connorjward  is it possible to block images from discussions? I can’t even read these screenshots because of the blurring.

**Reply by dham** [reply]
The problem is that there are legitimate use cases. E.g. posting a plot of a solution

**Comment by colinjcotter** [comment]
We probably just need a “nudge” in the issue template then.

On 2 Feb 2024, at 08:39, David A. Ham ***@***.***> wrote:

﻿

The problem is that there are legitimate use cases. E.g. posting a plot of a solution

**Comment by JDBetteridge** [comment, marked as answer]
As @connorjward  says please update your Firedrake install. You may still see errors, but they will not be fatal. (there's another other fix in the pipeline to make those go away)

**Reply by joernwichmann** [reply]
Indeed, the update solved the issue. Thanks for your help.
